backend/translation/views.py

```diff
@@ -182,18 +215,36 @@ def post(self, request):
                 status=status.HTTP_404_NOT_FOUND,
             )
 
-        # Return the appropriate response depending on whether a new translation was created or not
-        if created:
+        # If the translation belongs to the current user, update the translation
+        if translation.user == user:
+            translation.captions = captions
+            translation.translation_type=(
+                HUMAN_EDITED
+                if current_translation_type == MACHINE_GENERATED
+                else MANUALLY_CREATED
+            )
+            translation.save()
+            
             return Response(
-                {"message": "Translation created successfully."},
-                status=status.HTTP_201_CREATED,
+            {"message": "Translation updated successfully."}, status=status.HTTP_200_OK)
+    
+        else: 
+            new_translation = Translation.objects.create(
+                translation_type=MANUALLY_CREATED,
+                parent=translation,
+                transcript=translation.transcript,
+                target_lang=target_lang,
+                user=user,
+                payload=captions,
```

Comment:
@pranav2305 and I had a doubt about why this case was created. Is this the current functionality we wanted @GokulNC? This was created by Nishant so I am not sure why this was added originally.

Reply:
We have something similar in transcript as well:
https://github.com/AI4Bharat/Chitralekha-Backend/blob/master/backend/transcript/views.py#L355-L378

---

What this was supposed to mean is that, if the `translation.user is None`, it means that the translation is machine-generated. So we are supposed to consider it as a parent, and save the payload in a new child object.

So, the type here is supposed to be `HUMAN_EDITED`.

Reply:
And inside this `else`, please check if `translation.user == None`, and only if so, do this. If not, return an error saying that they don't have the rights to update someone else's translation.

Please handle this case in the transcript endpoint as well.

Reply:
This has been done for the `translation` app. Can you review the `transcript` app once because there a different logic is implemented for this case which was not done by me. (from lines 388 to 356)

Reply:
Uhh, what I meant for `transcript` app, is that, [in this block](https://github.com/AI4Bharat/Chitralekha-Backend/blob/70b74d7050149bb25edf0087b330c804877259d1/backend/transcript/views.py#L380-L383), before saving, we should check if `transcript.user == request.user`. If not, return an error saying that they are trying to write on someone else's transcript.

Reply:
Oh, okay. I got confused there. Sure, I will make that change.